apps/dashboard/spreadsheet.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+const XLSX = require('xlsx');
+const { flatMap, fromPairs } = require('lodash');
+
+function flattenApplicationData(application) {
+    return fromPairs(
+        flatMap(application.application_data, step => {
+            return flatMap(step.fieldsets, fieldset => {
+                return fieldset.fields.map(field => [field.name, field.value]);
+            });
+        })
+    );
+}
+
+function getTransformer(formId) {
+    switch (formId) {
+        case 'building-connections-fund':
+            return function(fields) {
+                return {
+                    'Project location': fields['location'],
+                    'Organisation Name': fields['organisation-name'],
+                    'Address: Street': fields['address-building-street'],
+                    'Address: Town or City': fields['address-town-city'],
+                    'Address: County': fields['address-county'],
+                    'Address: Postcode': fields['address-postcode']
+                };
+            };
+        default:
+            return null;
+    }
+}
+
+function summariseApplications({ baseUrl, formId, applications }) {
+    const transformer = getTransformer(formId);
+
+    if (transformer) {
+        return applications.map(application => {
+            const linkUrl = `${baseUrl}/${application.reference_id}`;
+            const coreFields = {
+                'Reference ID': application.reference_id,
+                'Full application URL': {
+                    f: `=HYPERLINK("${linkUrl}", "${linkUrl}")`
```

Comment:
Is this better off being a short code (for the link text) so the spreadsheet column can be smaller?